src/utils/tables.ts

```diff
@@ -0,0 +1,23 @@
+import { ProColumnType, TColumnStates } from '@ferlab/ui/core/components/ProTable/types';
+
+/*
+ * This function purpose is to determine if the columns in the user config
+ * match those of the table. If one changes a col key in a table than the
+ * user config, if used, could lead to incorrect display or action. Table cols must
+ * have priority.
+ * */
+const userColsHaveSameKeyAsDefaultCols = (
+  userCols: TColumnStates = [],
+  // must never ever be empty
+  defaultCols: ProColumnType[],
+): boolean => {
+  const userColsIsEmpty = !userCols || userCols.length === 0;
```

Comment:
Je pense tu peux réduire de même
```suggestion
  const userColsIsEmpty = userCols?.length === 0;
```